decode.go

```diff
@@ -446,6 +446,23 @@ func (fnbsm FieldNameByteStringMode) valid() bool {
 	return fnbsm >= 0 && fnbsm < maxFieldNameByteStringMode
 }
 
+// ReturnTypeForEmptyInterface specifies the type of the value to be returned when decoding into an empty interface
+type ReturnTypeForEmptyInterface int
+
+const (
+	// ReturnTagForEmptyInterface returns a value of type Tag
+	ReturnTagForEmptyInterface ReturnTypeForEmptyInterface = iota
+
+	// ReturnTagContentForEmptyInterface returns only the content of the Tag
+	ReturnTagContentForEmptyInterface
+
+	maxReturnTypesForEmptyInterface
+)
```

Comment:
Maybe rename and add more detailed comments.

```suggestion
// UnrecognizedTagToAnyMode specifies how to decode unrecognized CBOR tag into an empty interface (any).
// Currently recognized CBOR tag numbers are 0, 1, 2, 3, or registered by TagSet.
type UnrecognizedTagToAnyMode int

const (
	// UnrecognizedTagNumAndContentToAny decodes CBOR tag number and tag content to cbor.Tag 
	// when decoding unrecognized CBOR tag into an empty interface.
	UnrecognizedTagNumAndContentToAny UnrecognizedTagToAnyMode = iota

	// UnrecTagContentToAny decodes only CBOR tag content (into its default type) 
	// when decoding unrecognized CBOR tag into an empty interface.
	UnrecognizedTagContentToAny

	maxUnrecognizedTagToAny
)
```

Reply:
Thanks for the suggestion. I've made these changes.